entc/gen/template/dialect/sql/query.tmpl

```diff
@@ -249,11 +249,15 @@ func ({{ $receiver }} *{{ $builder }}) sqlCount(ctx context.Context) (int, error
 }
 
 func ({{ $receiver }} *{{ $builder }}) sqlExist(ctx context.Context) (bool, error) {
-	n, err := {{ $receiver }}.sqlCount(ctx)
-	if err != nil {
+	{{- if $.HasOneFieldID }}
+	_, err := {{ $receiver }}.FirstID(ctx)
+	{{- else }}
+	_, err := {{ $receiver }}.First(ctx)
+	{{- end }}
+	if err != nil && !IsNotFound(err) {
 		return false, fmt.Errorf("{{ $pkg }}: check existence: %w", err)
 	}
```

Comment:
```suggestion
	switch {
	case IsNotFound(err):
		return false, nil
	case err != nil:
		return false, fmt.Errorf("{{ $pkg }}: check existence: %w", err)
	default:
		return true, nil
	}
```

Reply:
Thanks @masseelch. I have addressed this and regenerated the files.